www/content/lexique.md

```diff
@@ -233,16 +243,18 @@ implémentations.
 keycode) est le code numérique issu du traitement bas niveau par le noyau, alors
 que le code _symbolique_ est le nom donné dans les fichiers XKB pour faciliter
 la configuration haut-niveau du clavier. On distinguera le code brut du _noyau_
-de celui de XKB, ce dernier étant obtenu en ajoutant 8 au premier.<br/>
-Exemple : sur un clavier QWERTY, le code brut noyau de la touche `Q` est
-`16`[^code-brut-noyau], le code brut sur XKB est `16 + 8 = 24` et son code
-symbolique est `AD01`[^code-symbolique-iso-9995].<br/>
-Noter qu’un clavier QWERTY et un clavier AZERTY produisent habituellement des
-keycodes identiques pour les touches situées au même emplacement : ainsi la
-touche imprimée `Q` sur le clavier QWERTY produira le même keycode que la touche
-imprimée `A` sur un clavier imprimé AZERTY. En effet, c’est la configuration du
-clavier dans XKB et non le clavier lui-même qui définit la correspondance touche/
-[keysym].
+de celui de XKB, ce dernier étant obtenu en ajoutant 8 au premier.
+
+    Exemple : sur un clavier QWERTY, le code brut noyau de la touche `Q` est
+    `16`[^code-brut-noyau], le code brut sur XKB est `16 + 8 = 24` et son code
+    symbolique est `AD01`[^code-symbolique-iso-9995].
```

Comment:
Je ne connaissais pas le truc pour faire des paragraphes dans les `<dl>`, bien vu. Le seul point qui me chagrine est que l’on ne les distingue pas des `<dd>`. Il faudrait sans doute gérer cela en CSS, en distinguant les cas où il y a un ou plusieurs `<dd>`. Visuellement j’aimerais éviter des puces ; à part mettre une bordure entre `<dd>` je n’ai pas d’idée pour l’instant.

Reply:
À voir si les `<dd>` multiples sont meilleurs que les parenthèses. Sémantiquement c’est mieux en tout cas.

Reply:
On peut faire vraiment très discret :
![image](https://github.com/Nuclear-Squid/ergol/assets/457778/2ad616da-52ce-4c15-917d-63fadac01a72)

Avec des puces on pourrait faire ça : 
![image](https://github.com/Nuclear-Squid/ergol/assets/457778/f9937ea1-5151-4ac9-865f-97bd32f2a2f1)

Les bordures horizontales, j’associe plutôt ça à des titres de niveau 2 ou 3, voire 4 pour du très fin, mais je trouve ça un peu bizarre à l’intérieur d’une section.

Reply:
Merci pour les tests. Du coup je préfère les puces. Tenter un `:not(:only-child)` pour que cela ne s’applique qu’aux listes avec au moins 2 définitions. Avec une seule ça va faire moche.

Reply:
Les bordures ne sont pas très visibles en mode sombre. Les puces sont discrètes et ont un bon contraste.

Reply:
On aurait pu peaufiner les styles pour les bordures mais de fait, les puces en demi-cadratin c’est bien plus simple. Go.

Reply:
Hmm, en fait le `:not(:only-child)` n’est pas une option parce que c’est une liste de définitions, pas une liste imbriquée. On peut sélectionner facilement la première définition avec `dt + dd` mais de là à détecter que c’est la seule définition, je ne sais pas.

Au pire ça ne me dérange pas plus que ça d’avoir une puce par définition, même unique.

Reply:
Oh, par contre c’est l’occasion de tester un `:has()` en condition réelle. Je creuse.

Reply:
… et je découvre `:only-of-type`, boulet que je suis…

Reply:
Je jette un coup d’œil après manger

Reply:
```css
  dd { margin: 0; padding: 0 1.5em; }
  dd:before {
    content: "–";
    float: left;
    text-align: center;
    width: 1.5em;
    margin-left: -1.5em;
  }
  dt + dd:last-child:before,
  dt + dd:has(+ dt):before {
    content: "";
  }
```

Il y a sûrement plus simple, mais ça fait le job.

Reply:
```css
  dd {
    display: list-item;
    list-style-type: "–  ";
  }
  dt + dd:last-child,
  dt + dd:has(+ dt) {
    list-style-type: "";
  }
```
plus simple

Reply:
On joue à du CSS golfing ?

```css
dt + dd:has(+ dd), dd + dd {
  display: list-item;
  list-style-type: disc;
}
```

Avantage : cela ne modifie pas le style de base, juste ce que l’on a besoin. Tu peux utiliser d’autres options pour [`list-style-type`](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type), notamment une chaîne de caractères. Si tu choisis cette option, il faut régler l’espace entre la puce et l’item. Le plus simple d’ajouter une espace : `list-style-type: "– ";`.

Reply:
Il y a plein de nouvelles options depuis que je me suis penché sérieusement sur le CSS. Les moteurs CSS vont être plus compliqués que les compilateurs de langages !

Reply:
Rhaaaaaa je déteste perdre au CSS golfing ou au Vim golfing ! Mais j’adopte. Merci !

Reply:
Marrant, je n’avais pas vu ta réponse et on arrive presque à la même conclusion.

Reply:
… y compris pour le hack de l’espace en fin de `list-style-type` :-)

Reply:
Les grands esprits se rencontrent ! 🥁